protocols/v2/codec-sv2/src/encoder.rs

```diff
@@ -1,3 +1,47 @@
+//! # Stratum V2 Codec Encoder
+//!
+//! This module provides functionality for encoding Stratum V2 messages, including support for the
+//! Noise protocol for secure communication.
+//!
+//! ## Features
+//!
+//! Two primary encoders are defined:
+//! * **Standard Encoder**:
+//! * **Noise Encoder**:
+//!
+//! ## Types
+//!
+//! * `Encoder`: Encoder for Sv2 frames with Noise protocol support.
+//! * `NoiseEncoder`: Encoder for Sv2 frames without Noise protocol support (requires the
+//!   `noise_sv2` feature).
+//! * `Item<T>`: Represents an SV2 frame that is to be encoded and optionally encrypted using the
+//!   Noise protocol.
+//! * `Buffer`: Holds intermediate data during encoding, which can be either a pool-allocated
+//!   buffer or a system memory buffer depending on the feature flags.
+//! * `Slice`: Stores the serialized SV2 frame data before transmission.
```

Comment:
no, we should not

also, the `## Types` section is redundant

when we render the docs via `cargo doc` command, all of those items are automatically rendered

IMHO we should avoid redundant docs as much as possible, because they can easily be overlooked when changes are applied to the code in the future, which will lead to outdated stuff.

https://github.com/stratum-mining/stratum/discussions/970#discussioncomment-10799306